### ISSUE TITLE
feature_request(ebooks): words with hyphens in the ends of lines

### ISSUE THREAD

**Issue by Kristinita** [phiresky/ripgrep-all#44]
### 1. Summary

It would be nice, if ripgrep-all will find words, which are separated by hyphens at the ends of lines.

![Okular expected behavior](https://i.imgur.com/kLzhf5R.png)

#### 2.1. Common cause

More correct search.

#### 2.2. Details

Hyphen between word parts used, for example, in [**English**](http://www.sussex.ac.uk/informatics/punctuation/hyphenanddash/hyphen), [**German**](https://www.duden.de/sprachwissen/rechtschreibregeln/worttrennung), [**Russian**](http://new.gramota.ru/biblio/readingroom/rules/141-perenos).

Currently, ripgrep-all doesn't find words, which are divided by hyphens. Users may not find the right words, because of what they may have problems with their work. The desired words can be separated by hyphens.

#### 2.3. Additional information

1. [**About optional hyphens**](https://practicaltypography.com/optional-hyphens.html)
1. [**Soft hyphen in PDF**](http://jkorpela.fi/shy.html#pdf)

#### 3.1. Data

1. [**`KiraTheEnigmaOfReason.pdf`**](https://app.box.com/s/qk1c1cqqqeb3w6tmnauq1i7pnooyngvq) — “[**The Enigma of Reason**](https://www.psychologytoday.com/us/blog/theory-knowledge/201808/the-enigma-reason-brief-review)” book.

#### 3.2. Purpose

I want to find word `understandable` on page 17 of this book. [**Okular**](https://okular.kde.org/) [**successfully found this word**](https://github.com/sumatrapdfreader/sumatrapdf/issues/1189):

![Okular expected behavior](https://i.imgur.com/kLzhf5R.png)

#### 3.3. Current behavior

```text
rga understandable
```

No results.

#### 3.4. Expected behavior

I converted `KiraTheEnigmaOfReason.pdf` to `KiraTheEnigmaOfReason.txt`:

```text
pdftotext KiraTheEnigmaOfReason.pdf
```

Now I run ripgrep-all:

```text
D:\SashaDebugging\KiraRipgrepAll>rga understandable
KiraTheEnigmaOfReason.txt
distrust. When we talk to ­others, we often have to overcome their understandable lack of trust. If we distrusted ­others only when they ­don’t deserve
```

pdftotext converted `under-standable` to `understandable`. From this I conclude, that problem in ripgrep-all, not in pdftotext.

### 4. Environment

1. Windows 10.0.18363 Pro N for Workstations 64-bit EN
1. ripgrep-all 0.9.3 (currently, [**the latest Windows version**](https://github.com/phiresky/ripgrep-all/issues/41))
1. pdftotext (from [**conda-forge Poppler**](https://anaconda.org/conda-forge/poppler/files)) 0.88.0
1. Okular 1.10.70

Thanks.

**Comment by phiresky** [comment]
Thanks for the sample file. Looks like the reason is that rga (which uses pdftotext internally) invokes it with the `-layout` option. `pdftotext -layout KiraTheEnigmaOfReason.pdf` gives the same result.

The reason it uses -layout is to make the text content correspond closer to the pdf page layout - but I didn't know this also meant reproducing hyphenation exactly:

> 
>       -layout
>              Maintain  (as  best  as  possible) the original
>              physical layout of the text.  The default is to
>              'undo'  physical  layout (columns, hyphenation,
>              etc.) and output the text in reading order.

Since we don't really care baout the layout much (except for page numbers), I think it's safe to just remove the `-layout` option.

**Comment by phiresky** [comment]
Fixed in 0.9.6

**Comment by Kristinita** [comment]
**Status: Confirmed** :heavy_check_mark:

Tested on example book and 3 modern books ([**1**](http://libgen.is/book/index.php?md5=06E62DA845897CA94A9B186A00C9AA5E), [**2**](http://libgen.is/book/index.php?md5=C1EE4D84053B0DD5D9DC6294B555C533), [**3**](http://libgen.is/book/index.php?md5=4D53C107BC614C717CCE8CF954512200)) with normal OCR layer.

Thanks.
